tensorflow/core/kernels/sequence_ops.cc

```diff
@@ -93,8 +95,21 @@ class RangeOp : public OpKernel {
     }
     int64_t size;
     if constexpr (std::is_integral<T>::value) {
-      size = Eigen::divup(Eigen::numext::abs(limit - start),
-                          Eigen::numext::abs(delta));
+      uint64_t range;
+      if ((limit > 0 && start < 0) || (limit < 0 && start > 0)) {
+        range = static_cast<uint64_t>(Eigen::numext::abs(limit))
+                + static_cast<uint64_t>(Eigen::numext::abs(start));
+      } else {
+        range = static_cast<uint64_t>(Eigen::numext::abs(limit - start));
+      }
+
+      uint64_t size_unsigned = Eigen::divup(range,
+                    static_cast<uint64_t>(Eigen::numext::abs(delta)));
+      OP_REQUIRES(context,
+                  size_unsigned <= std::numeric_limits<int64_t>::max(),
+                  InvalidArgument("Requires ((limit - start) / delta) <= ",
+                                  std::numeric_limits<int64_t>::max()));
```

Comment:
Does it _actually_ need to be less than int64_t max, or could it be uint64_t max (i.e. no real limit)?